shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs

```diff
@@ -474,6 +511,10 @@ public void RemoveAllCertificatesCore(IEnumerable<X509Certificate2> certificates
                 {
                     TrustCertificate(certificate);
                 }
+                catch(UserCancelledTrustException)
```

Comment:
Why do you need a custom exception type? Could you just change this to catch the CryptographicException here?

Reply:
There are a few reasons:
* I want to capture the exception at the OS level check to ensure that this is scoped to Windows.
* I didn't want to change trust certificate to return an error code.
* If we do something in the future for OSX the mechanism will probably be different, so the exception makes everything homogeneous.
* If we implement something like this for OSX the change can just be scoped to the OSX specific part.

Reply:
ok